### PR TITLE
SYNCOPE-1566: Manage GAuth Credentials/Accounts

### PR DESCRIPTION
**Pull request by mmoayyed**
- Provides endpoints, services and other components to manage WA GAuth accounts. Note that each account tied to an auth-profile in a one-to-one relationship. Supporting multiple devices per auth-profile can be an item to plan for in the future.

- Adds unit and integration tests

- Reorganizes the `wa-starter` module a bit to make sure components are moved to appropriate packages for audits and main config. 

- Renames a number of WA components to make sure they are prefixed with `SyncopeWA*` to be consistent with others.

### CODE REVIEW

**Comment by ilgrosso** [comment]
@mmoayyed it seems you have some trivial fixes to make, in order to allow ITs to succeed: https://travis-ci.org/github/apache/syncope/jobs/694721332#L737-L738

File: common/am/lib/src/main/java/org/apache/syncope/common/lib/types/GoogleMfaAuthAccount.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlType;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@XmlRootElement(name = "googleMfaAuthAccount")
```
Comment (ilgrosso) [inline common/am/lib/src/main/java/org/apache/syncope/common/lib/types/GoogleMfaAuthAccount.java:35]: Please remove all JAXB stuff

File: common/am/rest-api/src/main/java/org/apache/syncope/common/rest/api/service/wa/GoogleMfaAuthTokenService.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
 import org.apache.syncope.common.rest.api.RESTHeaders;
 import org.apache.syncope.common.rest.api.service.JAXRSService;
 
-@Tag(name = "Google MFA Tokens")
+@Tag(name = "Google MFA Accounts")
```
Comment (ilgrosso) [inline common/am/rest-api/src/main/java/org/apache/syncope/common/rest/api/service/wa/GoogleMfaAuthTokenService.java:45]: Are you sure about this change?!?

File: wa/starter/src/main/java/org/apache/syncope/wa/starter/config/SyncopeWAConfiguration.java [hunk 3]
```diff
@@ -7,7 +7,7 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *   http://www.apache.org/licenses/LICENSE-2.0
+ *    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (ilgrosso) [inline wa/starter/src/main/java/org/apache/syncope/wa/starter/config/SyncopeWAConfiguration.java:10]: I think this (and other similar) changes should be reverted.

File: wa/starter/src/main/java/org/apache/syncope/wa/starter/config/SyncopeWARefreshContextJob.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *   http://www.apache.org/licenses/LICENSE-2.0
+ *    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (ilgrosso) [inline wa/starter/src/main/java/org/apache/syncope/wa/starter/config/SyncopeWARefreshContextJob.java:10]: Watch license header formatting (also for new classes in this PR)

Comment (mmoayyed) [reply wa/starter/src/main/java/org/apache/syncope/wa/starter/config/SyncopeWARefreshContextJob.java]: I ran a search and replace in all offending components and hopefully should have fixed them all.

**Pull request merged**